soh/src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -5042,8 +4861,12 @@ s32 func_8083B040(Player* this, GlobalContext* globalCtx) {
                         func_80835DE4(globalCtx, this, func_8084F104, 0);
 
                         if (sp2C >= 0) {
-                            giEntry = &sGetItemTable[D_80854528[sp2C] - 1];
-                            func_8083AE40(this, giEntry->objectId);
+                            if (this->getItemEntry.objectId == OBJECT_INVALID) {
+                                giEntry = ItemTable_Retrieve(D_80854528[sp2C] - 1);
```

Comment:
another place that the `-1` is there, i'm still curious as to if removing that could clear up the logic or if it would make it more complex

Reply:
Same as other comments, - 1 is from vanilla due to GetItemIds being off by one from corresponding entries in sGetItemTable.

Reply:
Addressed by 69086f58c2b6bb00a273eefed4744ad456bba6ba